src/modules/tokens/services/token.compute.service.ts

```diff
@@ -551,32 +550,51 @@ export class TokenComputeService implements ITokenComputeService {
 
     async computeTokenLiquidityUSD(tokenID: string): Promise<string> {
         const pairs = await this.routerAbi.pairsMetadata();
-        const priceUSD = await this.tokenPriceDerivedUSD(tokenID);
-        const tokenMetadata = await this.tokenService.tokenMetadata(tokenID);
-        const promises = [];
-        for (const pair of pairs) {
-            switch (tokenID) {
-                case pair.firstTokenID:
-                    promises.push(this.pairAbi.firstTokenReserve(pair.address));
-                    break;
-                case pair.secondTokenID:
-                    promises.push(
-                        this.pairAbi.secondTokenReserve(pair.address),
-                    );
-                    break;
+
+        const relevantPairs = pairs.filter((pair) =>
+            [pair.firstTokenID, pair.secondTokenID].includes(tokenID),
```

Comment:
nitpicking, but a new array will be allocated at each iteration. either extract the array outside of the filter or refactor like:
tokenID === pair.firstTokenID || tokenID === pair.secondTokenID